### ISSUE TITLE
A warning about a brace or a semicolon behind a value with an inline comment is reported two columns late under `postcss-scss`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#139]
Split out of #115, which names the drift beside the fix that never arrives and repairs only the half of it a rule reads through the value utils.

`postcss-scss` rewrites every `//` comment of a value or of a set of parameters into a block comment inside `raws[prop].raw`, keeps the spelling of the file in `raws[prop].scss` and prints that second copy. `node.toString()` goes through the stringifier of PostCSS itself, which knows nothing of the second copy and prints the raw, so every position counted in that string stands two characters further along per comment than the file spells it — and an inline `Comment` node of its own drifts by the same two, since PostCSS prints it as a block comment as well.

A rule that locates a brace or a semicolon by the length of what a node prints therefore reports it in the wrong column.

### What minimal example or steps are needed to reproduce the bug?

```scss
a { b: 1px // c
  2px; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/block-closing-brace-newline-before": "always"
  }
}
```

### What did you expect to happen?

Spelled with a block comment of exactly the same width — `/**/` in place of `// c` — the same file draws the warning in column 7 of the second line. The inline comment changes nothing about where the brace stands, so the warning should not move.

### What actually happened?

The warning says column 9.

Six rules report a position of this kind: `at-rule-semicolon-space-before`, `block-closing-brace-newline-before`, `block-closing-brace-space-before`, `declaration-block-semicolon-newline-before`, `declaration-block-semicolon-space-before` and `declaration-block-trailing-semicolon`. Measured by linting a set of SCSS fixtures twice, once with an inline comment and once with a block comment of exactly the same length in its place, and keeping the pairs whose warnings disagree.

### Do you have a proposal to fix the bug?

The rules ask `node.toString()` for a length, not for a text, so what they need is the length of what the syntax would print. That cannot be had from the node alone: the difference lives in `raws[prop].scss` and in the `raws.inline` of every comment the node holds, and a node with a block carries both at any depth.

Two roads. A helper that walks the node and adds up the differences — for a value or a set of parameters, `scss.length - raw.length`; for an inline comment, whatever `//` spells against `/*…*/` — and hands back a corrected length. Or a stringifier of the plugin's own, mirroring the one each syntax installs, which every rule needing a printed node calls instead of `toString()`.

The first is smaller and answers exactly what these six rules ask. The second answers the general question, and would also serve a rule that needs the text rather than the length.

**Comment by firefoxic** [comment]
Two more findings for this issue, from an adversarial review of the branch that repairs #115.

The drift is not always only a column. On a query whose inline comment stands inside a media feature, the rules that measure a block by what a node prints raise or drop a warning outright:

```scss
@media (min-width: 100px // c
  ) { a { color: red; } }
```

With `block-opening-brace-space-after` set to `always`, that draws a warning at 2:2; the same file with `/**/` in place of `// c`, which is the same width, draws none. `block-opening-brace-space-after` (`always`, `always-multi-line`, `never-single-line`, `never-multi-line`), `block-opening-brace-space-before` (`never-single-line`, `never-multi-line`) and `block-closing-brace-space-before` (`never-single-line`, `never-multi-line`) all disagree with themselves that way.

The cause is the one this issue already names: what the rule reads is the raw, where the comment is a block one two characters longer, so the block string it measures is neither the length nor the text the file holds — and where the option turns on a block being single-line or multi-line, a difference of two characters is enough to change the answer rather than the column.

The second finding is a measurement. Linting a corpus of SCSS fixtures — every rule, every option, eighteen fixtures — twice each, once with an inline comment and once with a block comment of exactly the same width in its place, leaves 93 disagreements on the branch that repairs #115, against 118 on `main`. Every one of the 93 belongs here: eighty-five are a column, eight are the shape above. Nothing else in the plugin disagrees with itself over an inline comment any more.

**Comment by firefoxic** [comment]
A third measurement, from the control oracle written for the plan of 2026-08-20. It lints every fixture twice — once with a `//` comment and once with a block comment of exactly the same width — under every rule and every primary option, and keeps the pairs whose warnings disagree. It reports 114 rows over twelve rules on `main` at c47ab74.

Two things widen this issue.

**Twelve rules, not six.** Five of the six named above are among them; the sixth, `at-rule-semicolon-space-before`, no fixture of that corpus reaches yet, so it stands on the measurement this issue was filed with. The seven the oracle adds are `block-closing-brace-empty-line-before`, the four `block-opening-brace-*` rules, `indentation` and `unit-case`.

`indentation` and `unit-case` each have a defect of their own behind their rows — #236, #237, #233 and #234 — and neither of those is this one. Whether anything of this issue is left in the two once they are fixed wants measuring rather than assuming.

**`postcss-less` as well as `postcss-scss`.** The cause named above is the second copy `postcss-scss` keeps, and that syntax is the one the title says. The oracle disagrees under `postcss-less` too, where there is no second copy at all: a `//` comment is a `Comment` node carrying `raws.begin`, and PostCSS prints it as `/* c */`, three characters wider than the file spells it.

```less
a {
	color: pink;
	// c
}
```

`block-closing-brace-space-before` with `always` reports `4:2` for that file and `3:6` for the same file with a block comment of the same width in place of `// c`. Here it is the block comment the rule mis-places, not the inline one — the drift runs both ways, and only one of the two can be right.

That settles the choice of road the proposal above leaves open. A helper handing back a corrected length answers this issue for the rules that ask for a length; four of the twelve want the text — `blockString` slices one printed string by the length of another, `whitespaceChecker` reads characters out of the `source` it is handed, and `no-extra-semicolons` counts inside one — and a number leaves those four measuring one string and reading another, which is how #62 and #103 came about.

The spec is `.claude/specs/2026-08-20-fix-01-printed-node-length.md`.

**Comment by firefoxic** [comment]
A seventh rule reports a position of this kind, found by the review of the branch closing [#248](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/248): `declaration-block-semicolon-newline-after`, which places its warning by `decl.toString().length + 1` like the six above.

And it drifts further than a column. Where the value ends in the break that closes its own comment, the two characters the rewriting adds are enough to carry the index past the break itself, so the warning is reported on a line the file does not spell it on:

```scss
a { color: red // c
;
top: 0;
}
```

under `never-multi-line` the warning lands at `3:2`, where the semicolon stands at `2:1`. The same file with a block comment of exactly the same width, `/**/` in place of `// c`, draws it at `2:2`; `postcss-less`, which keeps one copy of the value, draws it at `2:2` as well.

So the census is seven rules rather than six, and the title of this issue is short by however many lines the drift can cross: the count is two characters per comment, and where those two characters stand across a line break they move the line as well as the column.

The branch closing #248 leaves this alone — its case asserts no position at all and says why — since repairing it is this issue's work.
